### ISSUE TITLE
Builds (especially release builds) are difficult to reproduce

### ISSUE THREAD

**Issue by bryteise** [kata-containers/packaging#735]
# Description of problem
I am trying to build a complete kata release from source (basically the content that kata project itself builds for its own binary releases). Ideally to do the builds I would have access to the complete sources used in a release tar for each project. Unfortunately some projects are git cloned during the build process and to make things more difficult they do not checkout a specific tag or even commit so they are very hard or impossible to replicate.

# Expected result
Each project has a single distribution tar with all required static dependencies included or at least has the tars for the required static dependencies available for each release.

**Comment by jodh-intel** [comment]
Hi @bryteise - Could you enumerate the projects that are causing problems? We use `dep` vendoring for golang, and everything else is *supposed* to be using either the master versions database:

- https://github.com/kata-containers/runtime/blob/93197ddcc3723e676d373959071b67385283f712/versions.yaml

... Or the tests versions database:

- https://github.com/kata-containers/runtime/blob/93197ddcc3723e676d373959071b67385283f712/versions.yaml

**Comment by bryteise** [comment]
@jodh-intel First I think that I just had changed some things in Clear Linux's build system and had made a mistake in my expectations with the vendoring used in 1.8.2. So this bug is invalid with respect to the go release tars though this repo is difficult for me to make use of for figuring out how to package kata. So what follows is a rambling attempt to do so that may prove useful or maybe at least useful to me in case you tell me I am doing stupid things and could do something better.

So I am going to try my hand at packaging kata from a different angle. First the https://katacontainers.io/software/ site, it has a broken link for the get the software pointing to https://git.katacontainers.io/cgit. Okay that's fine, I know the github repo obviously so I'll see what instructions exist for install via https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#initial-setup.

Since I need to build offline I grab the 1.8.2 release source tar for runtime and proceed on my merry way, success. Similarly I can build proxy and shim (this is where I somehow got stuck trying to use the packaging mechanism instructions for some reason and the reason for the bug).

I see some mention of initrd vs image. Clear Linux has been using https://github.com/kata-containers/agent/ releases up until this point rather than creating our own and I think this image's only real purpose is hosting the agent. So rather than trying to do a dance to build up an initrd and maybe have to patch runtime config I just fall back to the current method (should we change this?).

I'm skipping the kernel as we update that as part of a different process and it is really just the extra patch anyway.

Finally we get to qemu, the kata qemu-lite branch referenced in the documentation is very old. I know there have been multiple iterations on what to do here over the years so I'd like to try and adopt an approach that is both performant and maintained. The documentation doesn't really give me any ideas here though unfortunately. This repo has references to the stale qemu-lite branch and nemu. I have seen mention of using upstream qemu with patch(es) as well. For now I'm ignoring this and am just going to continue to use the old qemu-lite but an update to this section would be most welcome I think.

**Comment by jodh-intel** [comment]
Thanks for writing this up @bryteise.

# The issue with the katacontainers.io website

We no longer manage this site. The original source used to be https://github.com/kata-containers/www.katacontainers.io, but I believe it is currently generated from https://github.com/jimmytipit/kata-netlify-refresh.

We have requested that OSF move the content in that repo back to https://github.com/kata-containers/www.katacontainers.io.

# Using a rootfs image or an initrd image

I put this together to compare the two:

https://github.com/kata-containers/documentation/wiki/Kata-images

TL;DR - initrd is much faster, but less flexible.

# *emu's

Yeah, cough.. an embarrasment of riches :smile:

> This repo has references to the stale qemu-lite branch and nemu. 

Can you point to where you're seeing this?

>  an update to this section would be most welcome I think.

Again, if you could provide a pointer, we'll try to get this tidied up.

/cc @egernst.

**Comment by jodh-intel** [comment]
A quick update - the source for https://katacontainers.io/ is now (back) in https://github.com/kata-containers/www.katacontainers.io :smile:

**Comment by bryteise** [comment]
>> This repo has references to the stale qemu-lite branch and nemu.
> Can you point to where you're seeing this?
So I had thought I saw mention of nemu but I must have been paying attention to another document somewhere else. I was referring to [this](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#build-a-custom-qemu).
That uses [this](https://github.com/kata-containers/qemu/commits/qemu-lite-2.11.0) qemu-lite-2.11.0 branch which hasn't been updated since December of last year. Maybe that's right but it isn't something I'm going to trust at first glance.

Reading over the qemu configuration in that section, as a non-expert in qemu, I am left to try and see if I can reasonably merge the generated configuration from the kata-containers/packaging/scripts/configure-hypervisor.sh or just keep the two different qemu's around in the distro. Then I see things like [this](https://github.com/kata-containers/packaging/tree/master/qemu/patches/4.1.x) which point to upstream qemu needing to be patched anyway but that isn't mentioned in the developer guide. Given that the first recommendation is using a 2.11.0 fork I don't have high hopes for myself to be able to make something reasonable.
